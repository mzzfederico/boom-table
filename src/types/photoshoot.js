/* We can use these to state clearly what our components expect data to be formatted to. */
/* I'm also leaving here the toolset to mock them. Beware! */

const SIZES = [ "XS", "S", "M", "L", "XL", "XXL", "3XL" ];
const COUNTRIES = [ "ITALY", "FRANCE", "SPAIN", "GERMANY" ];
const WEEK_DAYS = [ "MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY" ];
const TYPES = [ "Food", "Events", "Real Estate" ];

function Photoshoot ({
    id = 0,
    day_of_the_week = WEEK_DAYS[(Math.floor(Math.random() * WEEK_DAYS.length))],
    type = TYPES[((Math.floor(Math.random() * TYPES.length)))],
    client_id = (Math.floor(Math.random() * 9) + 1),
    photoshoot_id = 0,
    details = new PhotoshootDetails({id: 0})
}) {
    return {
        id,
        day_of_the_week,
        type,
        client_id,
        photoshoot_id,
        details
    };
}

function PhotoshootDetails ({
    id = 0,
    title = "Test",
    number_of_photos = (Math.floor(Math.random() * 99) + 1),
    country = COUNTRIES[((Math.floor(Math.random() * COUNTRIES.length)))],
    package_size = SIZES[((Math.floor(Math.random() * SIZES.length)))]
}) {
    return {
        id,
        title,
        number_of_photos,
        country,
        "package": package_size
    };
}

function MockPhotoshoot(n) {
    return new Photoshoot({
        "id": n,
        "photoshoot_id": n,
        "details": new PhotoshootDetails({
            id: n
        })
    });
}

export {
    Photoshoot, PhotoshootDetails, MockPhotoshoot
};