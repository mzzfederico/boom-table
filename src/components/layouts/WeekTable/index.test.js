/* eslint-disable no-undef */
import React from "react";
import { WeekTable, PhotoshootCell, HeaderCell, PlainCell } from "./index";
import renderer from "react-test-renderer";
import { Photoshoot, PhotoshootDetails } from "../../../types/photoshoot";

it("WeekTable renders without crashing", () => {
    const tree = renderer.create(<WeekTable />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("PhotoshootCell renders without crashing", () => {
    const tree = renderer.create(
        <PhotoshootCell day={"MONDAY"} 
            photoshoot={new Photoshoot({
                id: 100,
                day_of_the_week: "MONDAY",
                type: "Food",
                client_id: "90",
                photoshoot_id: 100,
                details: new PhotoshootDetails({ 
                    id: 100,
                    title: "Test",
                    number_of_photos: 90,
                    country: "Italy",
                    package_size: "XL"
                })
            })}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it("HeaderCell renders without crashing", () => {
    const tree = renderer.create(<HeaderCell label={"Label test"} column={3} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("PlainCell renders without crashing", () => {
    const tree = renderer.create(<PlainCell day={"MONDAY"} variant={"total-top"} column={3} total={1000} />).toJSON();
    expect(tree).toMatchSnapshot();
});