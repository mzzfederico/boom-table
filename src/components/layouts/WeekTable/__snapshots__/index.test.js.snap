// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoshootCell renders without crashing 1`] = `
<div
  className="photoshoot photoshoot-monday"
>
  <span
    className="type"
  >
    Food
  </span>
  <span
    className="package"
  >
    XL
  </span>
  <span
    className="client"
  >
    90
  </span>
  <span
    className="number_of_photos"
  >
    n/a
  </span>
  <style
    jsx={true}
  >
    
            .WeekTable .photoshoot-monday {
                grid-column: 1
            }
        
  </style>
</div>
`;

exports[`WeekDayCol renders without crashing 1`] = `
Array [
  <div
    className="column column-monday"
  >
    monday
    <style
      jsx={true}
    >
      
                .WeekTable .column-monday {
                    grid-column: 1
                }
            
    </style>
  </div>,
  <div
    className="photoshoot photoshoot-monday"
  >
    <span
      className="type"
    >
      Food
    </span>
    <span
      className="package"
    >
      XL
    </span>
    <span
      className="client"
    >
      90
    </span>
    <span
      className="number_of_photos"
    >
      n/a
    </span>
    <style
      jsx={true}
    >
      
            .WeekTable .photoshoot-monday {
                grid-column: 1
            }
        
    </style>
  </div>,
]
`;

exports[`WeekTable renders without crashing 1`] = `
<div
  className="WeekTable"
>
  <div
    className="column column-monday"
  >
    monday
    <style
      jsx={true}
    >
      
                .WeekTable .column-monday {
                    grid-column: 1
                }
            
    </style>
  </div>
  <div
    className="column column-tuesday"
  >
    tuesday
    <style
      jsx={true}
    >
      
                .WeekTable .column-tuesday {
                    grid-column: 2
                }
            
    </style>
  </div>
  <div
    className="column column-wednesday"
  >
    wednesday
    <style
      jsx={true}
    >
      
                .WeekTable .column-wednesday {
                    grid-column: 3
                }
            
    </style>
  </div>
  <div
    className="column column-thursday"
  >
    thursday
    <style
      jsx={true}
    >
      
                .WeekTable .column-thursday {
                    grid-column: 4
                }
            
    </style>
  </div>
  <div
    className="column column-friday"
  >
    friday
    <style
      jsx={true}
    >
      
                .WeekTable .column-friday {
                    grid-column: 5
                }
            
    </style>
  </div>
  <div
    className="column column-saturday"
  >
    saturday
    <style
      jsx={true}
    >
      
                .WeekTable .column-saturday {
                    grid-column: 6
                }
            
    </style>
  </div>
  <div
    className="column column-sunday"
  >
    sunday
    <style
      jsx={true}
    >
      
                .WeekTable .column-sunday {
                    grid-column: 7
                }
            
    </style>
  </div>
  <style
    jsx={true}
  >
    
            .WeekTable {
                display: grid;
                grid-auto-flow: column;
                grid-template-columns: repeat(7, 1fr);
            }
        
  </style>
</div>
`;
